patches/tModLoader/Terraria/Recipe.TML.cs

```diff
@@ -19,32 +19,40 @@ namespace Terraria;
 /// </summary>
 public partial class Recipe
 {
+	[Obsolete($"Replaced by {nameof(IngredientQuantityRules)} due to not accounting for shimmer decrafting")]
 	public static class ConsumptionRules
 	{
-		/// <summary> Gives 1/3 chance for every ingredient to not be consumed, if used at an alchemy table. (!) This behavior is already automatically given to all items that can be made at a placed bottle tile. </summary>
-		public static ConsumeItemCallback Alchemy = (Recipe recipe, int type, ref int amount) => {
-			if (!Main.LocalPlayer.alchemyTable) return;
-
-			int amountUsed = 0;
+		[Obsolete($"Replaced by {nameof(IngredientQuantityRules)}.{nameof(IngredientQuantityRules.Alchemy)} due to not accounting for shimmer decrafting")]
+		public static ConsumeItemCallback Alchemy => (Recipe recipe, int type, ref int amount) => IngredientQuantityRules.Alchemy(recipe, type, ref amount, false);
+	}
 
-			for (int i = 0; i < amount; i++) {
-				if (!Main.rand.NextBool(3))
-					amountUsed++;
+	public static class IngredientQuantityRules
+	{
+		/// <summary> Gives 1/3 chance for every ingredient to not be consumed, if used at an alchemy table. (!) This behaviour is already automatically given to all items that can be made at a placed bottle tile. </summary>
+		public static IngredientQuantityCallback Alchemy = (Recipe recipe, int type, ref int amount, bool isDecrafting) => {
+			if (!Main.LocalPlayer.alchemyTable && !isDecrafting)
+				return;
+			for (int i = amount; i > 0; i--) {
+				if (Main.rand.NextBool(3))
+					amount--;
 			}
-
-			amount = amountUsed;
 		};
 	}
 
 	public readonly Mod Mod;
-	public readonly List<Condition> Conditions = new List<Condition>();
-	public readonly List<Condition> DecraftConditions = new List<Condition>();
+	public readonly List<Condition> Conditions = new();
+	public readonly List<Condition> DecraftConditions = new();
```

Comment:
if the code isn't necessary to change, let's try not to change it in future to keep review lite please.